ncc/examples/wu.c

```diff
@@ -0,0 +1,256 @@
+{
+    int dx = x1 - x0;
+    int dy = y1 - y0;
+    assert(dx > 0 && dy > 0 && dx > dy);
+    int error_adjust = LIMIT * dy / dx;
+    int error_accumulator = 0;
+    while (x1 >= x0)
+    {
+        u8 intensity = error_accumulator >> 8 & 0xff;
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x0, y0,     color, 0xFF - intensity);
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x0, y0 + 1, color, intensity);
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x1, y1,     color, 0xFF - intensity);
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x1, y1 - 1, color, intensity);
+        ++x0;
+        --x1;
+        error_accumulator = error_accumulator + error_adjust;
+        if (error_accumulator > LIMIT)
+        {
+            error_accumulator = error_accumulator - LIMIT;
+            ++y0;
+            --y1;
+        }
+    }
+}
+
+void draw_Wu_line_second_octant(u32* fb, u32 fb_width, u32 fb_height, u32 x0, u32 y0, u32 x1, u32 y1, u32 color)
+{
+    int dx = x1 - x0;
+    int dy = y1 - y0;
+    assert(dx > 0 && dy > 0 && dy > dx);
+    int error_adjust = LIMIT * dx / dy;
+    int error_accumulator = 0;
+    while (y1 >= y0)
+    {
+        u8 intensity = error_accumulator >> 8 & 0xff;
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x0, y0,     color, 0xFF - intensity);
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x0 + 1, y0, color, intensity);
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x1, y1,     color, 0xFF - intensity);
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x1 - 1, y1, color, intensity);
+        ++y0;
+        --y1;
+        error_accumulator = error_accumulator + error_adjust;
+        if (error_accumulator > LIMIT)
+        {
+            error_accumulator = error_accumulator - LIMIT;
+            ++x0;
+            --x1;
+        }
+    }
+}
+
+
+void draw_Wu_line_third_octant(u32* fb, u32 fb_width, u32 fb_height, u32 x0, u32 y0, u32 x1, u32 y1, u32 color)
+{
+    int dx = x1 - x0;
+    int dy = y0 - y1;
+    assert(dx > 0 && dy > 0 && dx > dy);
+    int error_adjust = LIMIT * dy / dx;
+    int error_accumulator = 0;
+    while (x1 >= x0)
+    {
+        u8 intensity = error_accumulator >> 8 & 0xff;
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x0, y0,     color, 0xFF - intensity);
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x0, y0 - 1, color, intensity);
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x1, y1,     color, 0xFF - intensity);
+        carefree_alpha_blend_plot_pixel(fb, fb_width, x1, y1 + 1, color, intensity);
+        ++x0;
+        --x1;
+        error_accumulator = error_accumulator + error_adjust;
+        if (error_accumulator > LIMIT)
+        {
+            error_accumulator = error_accumulator - LIMIT;
+            ++y1;
+            --y0;
+        }
+    }
+}
+
+void draw_Wu_line_fourth_octant(u32* fb, u32 fb_width, u32 fb_height, u32 x0, u32 y0, u32 x1, u32 y1, u32 color)
```

Comment:
nitpick but `_wu_` should be lowercase to match snake-case, even though it's a proper name.